From 2a50ed28e16df77f679c9f085eff41cb2f02c80f Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michaelni@gmx.at>
Date: Thu, 4 Aug 2011 01:04:50 +0200
Subject: [PATCH 2/2] avidec: skip seek pos adjustment for non interleaved files.
 Fixes Ticket327

Signed-off-by: Michael Niedermayer <michaelni@gmx.at>
---
 libavformat/avidec.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libavformat/avidec.c b/libavformat/avidec.c
index f71a270..92d8ff1 100644
--- a/libavformat/avidec.c
+++ b/libavformat/avidec.c
@@ -1371,7 +1371,7 @@ static int avi_read_seek(AVFormatContext *s, int stream_index, int64_t timestamp
                 flags | AVSEEK_FLAG_BACKWARD | (st2->codec->codec_type != AVMEDIA_TYPE_VIDEO ? AVSEEK_FLAG_ANY : 0));
         if(index<0)
             index=0;
-        while(index>0 && st2->index_entries[index-1].pos >= pos_min)
+        while(!avi->non_interleaved && index>0 && st2->index_entries[index-1].pos >= pos_min)
             index--;
         ast2->frame_offset = st2->index_entries[index].timestamp;
     }
-- 
1.7.4.1

