Index: libavutil/log.h
===================================================================
--- a/libavutil/log.h	(revision 30860)
+++ b/libavutil/log.h	(working copy)
@@ -140,11 +140,13 @@
  * Useful to print debug messages that shouldn't get compiled in normally.
  */
 
+#ifndef _MSC_VER
 #ifdef DEBUG
 #    define av_dlog(pctx, ...) av_log(pctx, AV_LOG_DEBUG, __VA_ARGS__)
 #else
 #    define av_dlog(pctx, ...) do { if (0) av_log(pctx, AV_LOG_DEBUG, __VA_ARGS__); } while (0)
 #endif
+#endif
 
 /**
  * Skip repeated messages, this requires the user app to use av_log() instead of
