diff -ru openssl-0.9.8y/util/pl/VC-32.pl openssl/util/pl/VC-32.pl
--- openssl-0.9.8y/util/pl/VC-32.pl	2013-02-05 11:58:59.000000000 +0000
+++ openssl/util/pl/VC-32.pl	2013-07-28 14:36:28.182738400 +0100
@@ -116,7 +116,7 @@
     $base_cflags.=' -D_CRT_NONSTDC_NO_DEPRECATE';	# shut up VC8
     my $f = $shlib || $fips ?' /MD':' /MT';
     $lib_cflag='/Zl' if (!$shlib);	# remove /DEFAULTLIBs from static lib
-    $opt_cflags=$f.' /Ox /O2 /Ob2';
+    $opt_cflags=$f.' /O1 /Ob1';
     $dbg_cflags=$f.'d /Od -DDEBUG -D_DEBUG';
     $lflags="/nologo /subsystem:console /opt:ref";
     }
