From 0c18eecd120fac527942cd8609545bfba7566b27 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 01:45:48 -0400
Subject: [PATCH 05/36] fixed: #5585 Media file gets played twice. we assume avi EOF when we can't find the next streams index entry for non interleaved files.

---
 libavformat/avidec.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

Index: ffmpeg/libavformat/avidec.c
===================================================================
--- ffmpeg.orig/libavformat/avidec.c	2011-08-07 07:06:24.399123200 +0100
+++ ffmpeg/libavformat/avidec.c	2011-08-07 07:06:26.792564800 +0100
@@ -1050,7 +1050,8 @@
         if(i>=0){
             int64_t pos= best_st->index_entries[i].pos;
             pos += best_ast->packet_size - best_ast->remaining;
-            avio_seek(s->pb, pos + 8, SEEK_SET);
+            if(avio_seek(s->pb, pos + 8, SEEK_SET) < 0)
+              return AVERROR_EOF;
 //        av_log(s, AV_LOG_DEBUG, "pos=%"PRId64"\n", pos);
 
             assert(best_ast->remaining <= best_ast->packet_size);
@@ -1060,6 +1061,8 @@
                 best_ast->packet_size=
                 best_ast->remaining= best_st->index_entries[i].size;
         }
+        else
+          return AVERROR_EOF;
     }
 
 resync:
