From d684ae10f93158ff4497187cc70bcaa2257a8639 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 01:55:31 -0400
Subject: [PATCH 07/36] if av_read_packet returns AVERROR_IO, we are done. ffmpeg's codecs might or might not handle returning any completed demuxed packets correctly

---
 libavformat/utils.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Index: ffmpeg/libavformat/utils.c
===================================================================
--- ffmpeg.orig/libavformat/utils.c	2011-11-13 17:44:59.299931700 +0000
+++ ffmpeg/libavformat/utils.c	2011-11-13 18:28:47.747849400 +0000
@@ -1219,6 +1219,8 @@
             if (ret < 0) {
                 if (ret == AVERROR(EAGAIN))
                     return ret;
+                if (ret == AVERROR(EIO))
+                    return ret;
                 /* return the last frames, if any */
                 for(i = 0; i < s->nb_streams; i++) {
                     st = s->streams[i];
