From d6d3554d4c312a739b71ac67db594fde2e297d46 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 02:10:50 -0400
Subject: [PATCH 08/36] added: Ticket #7187, TV Teletext support for DVB EBU Teletext streams

---
 libavcodec/avcodec.h |    4 ++++
 libavformat/mpegts.c |    2 ++
 2 files changed, 6 insertions(+), 0 deletions(-)

Index: ffmpeg/libavcodec/avcodec.h
===================================================================
--- ffmpeg.orig/libavcodec/avcodec.h	2011-08-07 07:06:17.293848600 +0100
+++ ffmpeg/libavcodec/avcodec.h	2011-08-07 07:06:28.014321600 +0100
@@ -362,6 +362,10 @@
     CODEC_ID_SRT,
     CODEC_ID_MICRODVD,
 
+    /* data codecs */
+    CODEC_ID_VBI_DATA= 0x17500,
+    CODEC_ID_VBI_TELETEXT,
+
     /* other specific kind of codecs (generally used for attachments) */
     CODEC_ID_TTF= 0x18000,
 
Index: ffmpeg/libavformat/mpegts.c
===================================================================
--- ffmpeg.orig/libavformat/mpegts.c	2011-08-07 07:06:17.303848600 +0100
+++ ffmpeg/libavformat/mpegts.c	2011-08-07 07:06:28.024336000 +0100
@@ -563,6 +563,8 @@
     { 0x7b, AVMEDIA_TYPE_AUDIO,             CODEC_ID_DTS },
     { 0x56, AVMEDIA_TYPE_SUBTITLE, CODEC_ID_DVB_TELETEXT },
     { 0x59, AVMEDIA_TYPE_SUBTITLE, CODEC_ID_DVB_SUBTITLE }, /* subtitling descriptor */
+    { 0x45, AVMEDIA_TYPE_DATA,         CODEC_ID_VBI_DATA }, /* VBI Data descriptor */
+    { 0x46, AVMEDIA_TYPE_DATA,     CODEC_ID_VBI_TELETEXT }, /* VBI Teletext descriptor */
     { 0 },
 };
 
