From 45f65bca80b6d4cfd52a141c0d666c5050419960 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 02:40:23 -0400
Subject: [PATCH 10/36] don't reparse PMT unless it's version has changed readded: don't accept PMT that isn't current

---
 libavformat/mpegts.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

Index: ffmpeg/libavformat/mpegts.c
===================================================================
--- ffmpeg.orig/libavformat/mpegts.c	2011-08-07 07:06:28.545084800 +0100
+++ ffmpeg/libavformat/mpegts.c	2011-08-07 07:06:29.015761600 +0100
@@ -73,6 +73,7 @@
 struct MpegTSFilter {
     int pid;
     int last_cc; /* last cc code (-1 if first packet) */
+    int last_version; /* last version of data on this pid */
     enum MpegTSFilterType type;
     union {
         MpegTSPESFilter pes_filter;
@@ -327,6 +328,7 @@
     filter->type = MPEGTS_SECTION;
     filter->pid = pid;
     filter->last_cc = -1;
+    filter->last_version = -1;
     sec = &filter->u.section_filter;
     sec->section_cb = section_cb;
     sec->opaque = opaque;
@@ -429,6 +431,7 @@
     uint8_t tid;
     uint16_t id;
     uint8_t version;
+    uint8_t current;
     uint8_t sec_num;
     uint8_t last_sec_num;
 } SectionHeader;
@@ -500,6 +503,7 @@
     val = get8(pp, p_end);
     if (val < 0)
         return -1;
+    h->current = val & 0x1;
     h->version = (val >> 1) & 0x1f;
     val = get8(pp, p_end);
     if (val < 0)
@@ -1180,6 +1184,12 @@
         return;
     if (h->tid != PAT_TID)
         return;
+    if (!h->current)
+        return;
+    if (h->version == filter->last_version)
+        return;
+    filter->last_version = h->version;
+    av_dlog(ts->stream, "version=%d\n", filter->last_version);
 
     ts->stream->ts_id = h->id;
 
