From d3fb4761b262301398e013c703382d3f54f7a1f5 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 22:46:49 -0400
Subject: [PATCH 17/36] Don't assume reading more frames in av_find_stream_info will change a CODEC_ID_NONE into something better

---
 libavformat/utils.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: ffmpeg/libavformat/utils.c
===================================================================
--- ffmpeg.orig/libavformat/utils.c	2011-11-13 18:23:49.414161300 +0000
+++ ffmpeg/libavformat/utils.c	2011-11-13 18:28:45.783849500 +0000
@@ -2194,7 +2194,7 @@
         val = 1;
         break;
     }
-    return avctx->codec_id != CODEC_ID_NONE && val != 0;
+    return avctx->codec_id != CODEC_ID_PROBE && val != 0;
 }
 
 static int has_decode_delay_been_guessed(AVStream *st)
