From d4a0766af6b82ab3c99a56c726ec66cada0da6de Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Fri, 9 Jul 2010 15:21:10 -0400
Subject: [PATCH 19/36] changed: check for seek error when attempting to read timestamp in rm demuxer

---
 libavformat/rmdec.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

Index: ffmpeg/libavformat/rmdec.c
===================================================================
--- ffmpeg.orig/libavformat/rmdec.c	2011-08-07 07:06:16.793848600 +0100
+++ ffmpeg/libavformat/rmdec.c	2011-08-07 07:06:31.839822400 +0100
@@ -905,7 +905,9 @@
     if(rm->old_format)
         return AV_NOPTS_VALUE;
 
-    avio_seek(s->pb, pos, SEEK_SET);
+    if (avio_seek(s->pb, pos, SEEK_SET) < 0)
+        return AV_NOPTS_VALUE;
+
     rm->remaining_len=0;
     for(;;){
         int seq=1;
