--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -703,5 +703,5 @@
             if(av_log2(pd->buf_size) != av_log2(pd->buf_size - pkt->size)){
                 //FIXME we dont reduce score to 0 for the case of running out of buffer space in bytes
-                set_codec_from_probe_data(s, st, pd, st->probe_packets > 0 ? AVPROBE_SCORE_MAX/4 : 0);
+                set_codec_from_probe_data(s, st, pd, st->probe_packets > 0 ? AVPROBE_SCORE_MAX/4-1 : 0);
                 if(st->codec->codec_id != CODEC_ID_PROBE){
                     pd->buf_size=0;
