Index: ffmpeg/libavutil/common.h
===================================================================
--- ffmpeg.orig/libavutil/common.h	2011-11-13 17:45:00.003931700 +0000
+++ ffmpeg/libavutil/common.h	2011-11-13 19:40:58.585132900 +0000
@@ -61,6 +61,10 @@
 #define FF_ARRAY_ELEMS(a) (sizeof(a) / sizeof((a)[0]))
 #define FFALIGN(x, a) (((x)+(a)-1)&~((a)-1))
 
+#if defined(_MSC_VER) && ! defined(UINT64_C)
+  #define UINT64_C(val) val##ui64
+#endif
+
 /* misc math functions */
 extern const uint8_t ff_log2_tab[256];
 
