Only in /home/jools/Repos/Public/rtmpdump/librtmp: amf.o
diff -ru /home/jools/Repos/Public/rtmpdump/librtmp/handshake.h ./handshake.h
--- /home/jools/Repos/Public/rtmpdump/librtmp/handshake.h	2010-04-29 18:34:56.037549000 +0100
+++ ./handshake.h	2010-05-20 00:00:30.241412000 +0100
@@ -422,7 +422,7 @@
 #else
   ip = (int32_t *)(clientsig+8);
   for (i = 2; i < RTMP_SIG_SIZE/4; i++)
-    *ip++ = rand();
+    *ip++ = (rand()<<16) & rand();
 #endif
 
   /* set handshake digest */
Only in /home/jools/Repos/Public/rtmpdump/librtmp: hashswf.o
Only in /home/jools/Repos/Public/rtmpdump/librtmp: librtmp.a
Only in .: librtmp.diff
Only in .: libRTMP.vcproj
Only in /home/jools/Repos/Public/rtmpdump/librtmp: log.o
Only in /home/jools/Repos/Public/rtmpdump/librtmp: parseurl.o
Only in .: readme.txt
Only in .: Release
Only in /home/jools/Repos/Public/rtmpdump/librtmp: rtmp.o
diff -ru /home/jools/Repos/Public/rtmpdump/librtmp/rtmp_sys.h ./rtmp_sys.h
--- /home/jools/Repos/Public/rtmpdump/librtmp/rtmp_sys.h	2010-05-11 14:28:58.121802099 +0100
+++ ./rtmp_sys.h	2010-05-20 00:00:30.221413000 +0100
@@ -26,6 +26,11 @@
 #ifdef _XBOX
 #include <xtl.h>
 #include <winsockx.h>
+#include "..\..\cores\DllLoader\exports\emu_socket\emu_socket.h"
+struct mphostent* __stdcall dllgethostbyname(const char* name);
+#define gethostbyname(name) dllgethostbyname(name)
+#define hostent mphostent
+#define h_addr h_addr_list[0]
 #define snprintf _snprintf
 #define strcasecmp stricmp
 #define strncasecmp strnicmp
Only in /home/jools/Repos/Public/rtmpdump/librtmp: .svn
