From 470e2804c01b609532d95539c66ebb6d13c7f805 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Fri, 9 Jul 2010 17:36:00 -0400
Subject: [PATCH 27/36] changed: check return value of seeks to avoid messing up current stream location

---
 libavformat/avidec.c |   16 ++++++++++++----
 1 files changed, 12 insertions(+), 4 deletions(-)

Index: ffmpeg/libavformat/avidec.c
===================================================================
--- ffmpeg.orig/libavformat/avidec.c	2011-05-27 18:12:50.000000000 +0100
+++ ffmpeg/libavformat/avidec.c	2011-05-27 18:28:16.139101330 +0100
@@ -224,13 +224,18 @@
                 return -1;
             }
 
-            avio_seek(pb, offset+8, SEEK_SET);
+            if(avio_seek(pb, offset+8, SEEK_SET) < 0)
+                return -1;
             avi->odml_depth++;
             read_braindead_odml_indx(s, frame_num);
             avi->odml_depth--;
             frame_num += duration;
 
-            avio_seek(pb, pos, SEEK_SET);
+            if(avio_seek(pb, pos, SEEK_SET) < 0) {
+                av_log(s, AV_LOG_ERROR, "Failed to restore position after reading index");
+                return -1;
+            }
+
         }
     }
     avi->index_loaded=1;
@@ -1329,11 +1334,13 @@
         /* the av_index_search_timestamp call above.                     */
         assert(stream_index == 0);
 
+        if(avio_seek(s->pb, pos, SEEK_SET) < 0)
+            return -1;
+
         /* Feed the DV video stream version of the timestamp to the */
         /* DV demux so it can synthesize correct timestamps.        */
         dv_offset_reset(avi->dv_demux, timestamp);
 
-        avio_seek(s->pb, pos, SEEK_SET);
         avi->stream_index= -1;
         return 0;
     }
@@ -1384,7 +1391,8 @@
     }
 
     /* do the seek */
-    avio_seek(s->pb, pos_min, SEEK_SET);
+    if(avio_seek(s->pb, pos, SEEK_SET) < 0)
+        return -1;
     avi->stream_index= -1;
     return 0;
 }
