Index: ffmpeg/libavutil/log.h
===================================================================
--- ffmpeg.orig/libavutil/log.h	2011-11-13 19:00:33.553558100 +0000
+++ ffmpeg/libavutil/log.h	2011-11-13 19:00:38.961334100 +0000
@@ -152,11 +152,13 @@
  * Useful to print debug messages that shouldn't get compiled in normally.
  */
 
+#ifndef _MSC_VER
 #ifdef DEBUG
 #    define av_dlog(pctx, ...) av_log(pctx, AV_LOG_DEBUG, __VA_ARGS__)
 #else
 #    define av_dlog(pctx, ...) do { if (0) av_log(pctx, AV_LOG_DEBUG, __VA_ARGS__); } while (0)
 #endif
+#endif
 
 /**
  * Skip repeated messages, this requires the user app to use av_log() instead of
